packages/grafana-alerting/src/grafana/matchers/utils.ts

```diff
@@ -5,8 +5,16 @@ import { Label, LabelMatcher } from './types';
 // LabelMatchDetails is a map of labels to their match results
 export type LabelMatchDetails = {
   labelIndex: number; // index of the label in the labels array
-  match: boolean;
-  matcher: LabelMatcher | null;
+} & (PositiveLabelMatch | NegativeLabelMatch);
```

Comment:
Splitting this into a union type allows the type checker to infer that `matcher == null` only when `match == false` 👍